PyInstaller/utils/hooks/conda.py

```diff
@@ -0,0 +1,196 @@
+#
+# Distributed under the terms of the GNU General Public License (version 2
+# or later) with exception for distributing the bootloader.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+#
+# SPDX-License-Identifier: (GPL-2.0-or-later WITH Bootloader-exception)
+#-----------------------------------------------------------------------------
+
+# language=rst
+"""
+Additional helper methods for working specifically with Anaconda distributions
+are found at :attr:`PyInstaller.util.hooks.conda_support`. These functions find
+and parse the distribution metadata from json files located in the
+``conda-meta`` directory.
+
+.. versionadded:: 4.1.0
+
+This module is available only if run inside a Conda environment. Usage of this
+module should therefore be guarded by a try/except clause::
+
+    try:
+        from PyInstaller.utils.hooks import conda_support
+    except ImportError:
+        # Not using conda.
+        pass
+    else:
+        # Code goes here. e.g.
+        binaries = conda_support.collect_dynamic_libs("numpy")
+        ...
+
+Packages are all referenced by the *distribution name* you use to install it,
+rather than the *package name* you import it with. i.e Use
+``search_distribution("pillow")`` instead of ``search_distribution("PIL")``.
+
+"""
+
+import sys
+from pathlib import Path
+import json
+
+from PyInstaller import compat
+from PyInstaller.log import logger
+
+
+# Conda virtual environments each get their own copy of `conda-meta` so the use
+# of `sys.prefix` instead of `sys.base_prefix`, `sys.real_prefix`  or anything
+# from our `compat` module is intentional.
+CONDA_ROOT = Path(sys.prefix)
+CONDA_META_DIR = CONDA_ROOT / "conda-meta"
+
+
+def search_distribution(name):
+    """
+     Finds and extracts the distribution metadata from the distribution json
+     files in the ``conda-meta`` directory.
+
+    :param name: Distribution name.
+    :type name: str
+    :return: The json-parsed distribution information.
+    :rtype: dict
+    :raises: ModuleNotFoundError: If **distribution** was not installed with
+             conda.
+
+    This function collects a single distribution. But this is generally
+    discouraged due to Conda's often moving distribution contents to
+    base-distributions. e.g. *numpy* itself is empty but depends on
+    *numpy-base*, which contains the python files and depends on *mkl* and
+    *blas* which contain the DLLs. Instead use :meth:`walk_dependency_tree`.
+
+    """
+    # Locate the right json. The name format is:
+    #   [distribution name]-[version][ugly build # hash].json
+
+    # The json name can't be determined exactly so they must be glob searched:
+    for path in CONDA_META_DIR.glob(name + "-*.json"):
+        # But this will include distributions with a prefix of 'name'.
+        # e.g. 'numpy' would also pick up 'numpy-base'.
+        dic = json.loads(path.read_text())
+        if name == dic["name"]:
```

Comment:
Just a slight problem...
```suggestion
        if name == dic[name]:
```

Reply:
No, actually we do want to check if the `"name"` attribute in the json matches the `name` we are looking for. So the quotes are deliberate. I'll chuck in a comment.